imap/sync_server.c

```diff
@@ -395,6 +395,9 @@ static void usage(void)
  */
 void shut_down(int code)
 {
+    /* run any delayed commands before we go into shutdown */
+    libcyrus_run_delayed();
+
     in_shutdown = 1;
```

Comment:
Like this one is careful to not set `in_shutdown = 1` until after the libcyrus_run_delayed, and even has a comment about it

Reply:
Yeah, I reverted that behaviour - and I'll drop this.

The `in_shutdown` flag says "finish fast", so repacking is a bad choice then!  It happens because we've had a sigquit.